### ISSUE TITLE
Ads appearance enhancement

### ISSUE THREAD

**Issue by LinkStatu** [shubham-gupta-16/SimpleWallpaperApp-Android#4]
I have several suggestions regarding **Ads** and hope you consider them:

**1-** Replacing **Splash interstitial Ad** with "**App Open Ad**", Because interstitial Ad at App opening may be considered a violation to Google policies.

**2-** Replacing **Smart Banner Ad** with **Normal Banner** and add it to these sections:
- Home page (Wallpapers page)
- Collections page
- Favorites page
- Wallpaper activity (when the user open a single wallpaper, a **Banner Ad** will show inside it)

Usually **Banner Ad** doesn't bother the user app experience, so adding it to these section would be good for the App.

**3-** For **Premium wallpapers**, I noticed when I apply a wallpaper, there are 2 Ads will show, **Reward Ad &  interstitial Ad** and that would be somewhat annoying for user experience, so I suggest removing **interstitial Ad** in **Premium wallpapers**.

**4-** I suggest Implementing **interstitial Ad** when the user saves a non-premium wallpaper. In that way, we will have 2 **interstitial Ads**, when the user saves non-premium wallpapers and when he applies them.

**5-** I suggest Implementing **interstitial Ad** after a certain number of clicks, lets say 7 clicks. Let me explain:

- When the user clicks on any wallpaper (whether it is a **premium wallpaper** or not), this is counted as 1 click
- When the user clicks on any collection, this is counted as 1 click

Lets say now he has 6 clicks, when he does the 7th click, an **interstitial Ad** will show.
I believe in that way, the user will get a good App experience and in the same time, a proper number of Ads will show. Hope you consider my suggestions and thanks a lot for your efforts.

**Comment by shubham16g** [comment]
@LinkStatu Thank you for your feedback. After reading the Admob policies, I found that this app doesn't follow them correctly. So, I made some changes to your suggestions and came up with these solutions respectively.

1. Splash Interstitial Ad will be removed.

2. I found that Admob discourages using SMART_BANNER and suggests using Anchored Adaptive Banners. Either a Normal Banner or an Anchored Adaptive Banner will be used. I will decide after testing them.

I will add the Banner ad on HomeScreen above the Bottom-Nav and SetWallpaperScreen.

3. Interstitial Ad for Premium Wallpapers will be removed.

4. Interstitial Ad on saving Non-Premium Wallpapers will be added.

5. Interstitial Ad after certain clicks will be implemented. But, instead of showing them on open the FullScreenWallpaperScreen, I suggest implementing it on closing the screen (on Back Pressed). This will not be shown for premium wallpapers or a non-premium wallpaper is applied or saved (Interstitial Already Showed).

I will try to implement these as soon as possible. If you have any other suggestions or found any mistakes in the above points, let me know. I will happy to solve these issues.

**Comment by LinkStatu** [comment]
Ok, that's will be great.
What about "**App Open Ad**", will you implement it too?

**Comment by shubham16g** [comment]
> Ok, that's will be great. What about "**App Open Ad**", will you implement it too?

https://developers.google.com/admob/android/app-open

Hmm, this looks fine. I will implement it too.

**Comment by LinkStatu** [comment]
Great!
I forgot to mention something else. For bottom-nav inside the wallpaper itself, it is very close to the Bottom navigation bar of Android. Here is what I mean: [Screen](https://images2.imgbox.com/39/f7/Wl4T3VVw_o.jpg)

Idk if it's just for me or it looks like that for all

**Comment by shubham16g** [comment]
> Great!
> I forgot to mention something else. For bottom-nav inside the wallpaper itself, it is very close to the Bottom navigation bar of Android. Here is what I mean: [Screen](https://images2.imgbox.com/39/f7/Wl4T3VVw_o.jpg)
> 
> Idk if it's just for me or it looks like that for all
> 

Don't worry I will fix it too.

**Comment by shubham16g** [comment]
@LinkStatu I implemented the changes. Test it and share your feedback. I would be glad to improve the code.

**Comment by LinkStatu** [comment]
@shubham-gupta-16 Thank you so much for your hard working. I can tell, now, it's the best wallpaper app ever 😁.

> Test it and share your feedback

Actually, I did, yesterday, and found a bug in "**Collections**" page. When I enter a collection and get out to enter another one, the app keep stopping or taking me to the home page.

I recorded a video about that. watch it from [Here](https://sendvid.com/zq1qtlq2?secret=94f84510-b72b-496c-a7a1-67185c31ce25)
